### ISSUE TITLE
James's shadow / Lowres fonts

### ISSUE THREAD

**Issue by Godzie** [elishacloud/Silent-Hill-2-Enhancements#34]
Good day folks! First, I want to thank you for this lot of work!
I've got a few questions and maybebugs about the mod.
1)For now game just ignores screen resolution from advanches option?
2)Modification should be installed on basic eu/na version of the game? Without any official patches(I can't remember is there any)?
2)James's shadow in first location looks really weird. Especially in motion:
![shadow01](https://user-images.githubusercontent.com/42905455/44953983-d9c29b00-aec5-11e8-8707-1cba5369bc66.jpg)
3)About the game's font. Picture says more than words:
![sh2pc 2018-09-02 14-18-16](https://user-images.githubusercontent.com/42905455/44954072-ad0f8300-aec7-11e8-862b-befd446823d3.jpg)

**Comment by Polymega** [comment]
1) Correct
2A) Correct
2B) This is how his shadow is supposed to look whenever James is outside during the day. The PS2 version renders the shadow the same way.
3) This is normal. We plan to fix this sometime in the future. You can set `Fix2D = 0` and `FullscreenImages = 0` in the `d3d8.ini` file to stretch the images for the time being.

**Comment by Godzie** [comment]
Oh, the numbers...
2B)If I correct [HD Collection](https://youtu.be/wh6LhkA3698?t=451) fix that. Or they just lower light intensity?
3)Sorry for misunderstanding, I didn't mean grey borders there, I have no problem with borders when they didn't mess aspect ratio. I was talking about low resolution font:
![lowpoly](https://user-images.githubusercontent.com/42905455/44957968-51fb8180-af03-11e8-9f7d-e13cd62c0112.jpg)

**Comment by Polymega** [comment]
We're not sure if it's possible yet but we definitely want to implement HD fonts, if it's possible.

**Comment by AeroWidescreen** [comment]
> For now game just ignores screen resolution from advanches option?

With the widescreen fix enabled, yes. You can use whatever resolution your hardware supports, but you have to change it through the d3d8.ini. Look for "ResX" and "ResY'.

> 2)James's shadow in first location looks really weird. Especially in motion:

Anytime you're indoors or it's dark outside the shadows look much better. Basically anytime there's no fog. It was probably done to help performance if I had to take a guess.

**Comment by Polymega** [comment]
> For now game just ignores screen resolution from advanches option?

We also have an idea to have the in-game Options menu show the resolution you set in the `d3d8.ini` file (and also prevent you from changing this resolution using the in-game Options menu). But because this would be such minor adjustment, it's low on the list to implement for the time being.

**Comment by Adam-Saeed** [comment]
Sorry to bother you guys, but I was wondering if there's a way to decrease the spacing between the letters.
![image](https://user-images.githubusercontent.com/43554017/47118792-27764580-d271-11e8-8c75-a4c5c6b73613.png)
(the picture is from SH2 HD Collection, but still works as a valid example)

**Comment by Polymega** [comment]
Hi @Adam-Saeed , 

> I was wondering if there's a way to decrease the spacing between the letters.

I'm not immediately sure if that is possible to do or not. @lakeviewhotel recently translated this game to Chinese and @belek666 has made a SH2/3/4 PC font extractor found here: https://github.com/belek666/fontsh234

Perhaps one of them might be able to answer if it's possible to adjust letter spacing or not.

**Comment by elishacloud** [comment]
Release an update with this fix in it.  See [v1.3.1340.0](https://github.com/elishacloud/Silent-Hill-2-Enhancements/releases/tag/v1.3.1340.0)

**Comment by AloisManas** [comment]
Hello,
I've encountered the same shadow issue as the OP in the SH2 EE PC version.

This issue however does not exist when I play the game on PCSX2 and I believe it didn't exist when I played the game on an actual PS2, though I'm unable to verify that at the moment.

Here are screenshots from PCSX2 (pcsx2-v1.5.0-dev-2855-g781323ff1-windows-x86), using the OGL Hardware renderer:
![sh2_pcsx2_01](https://user-images.githubusercontent.com/53020595/61410399-79bdc300-a8e4-11e9-9309-da72d2060597.png)
![sh2_pcsx2_02](https://user-images.githubusercontent.com/53020595/61410428-89d5a280-a8e4-11e9-9e6b-65e1bdef4c6a.png)

**Comment by Polymega** [comment]
Interesting. I'm using 1.5.0 dev 2361 (older version) and I get this, which is the same as the PC version: 

![image](https://user-images.githubusercontent.com/27395260/61412511-136e7b80-a8b7-11e9-88a2-4469d56002aa.png)
![image](https://user-images.githubusercontent.com/27395260/61412637-5cbecb00-a8b7-11e9-87f1-503c95b19a6f.png)

I'll try and test with real PS2 hardware soon and let you know what I find there.

**Comment by AloisManas** [comment]
This issue seems to relate to the Accurate Date setting in the PCSX2 OGL Hardware render settings. When I set this to Off, I have the same ugly shadows like on your screenshots. But when I set this to Full the shadows look correctly.

**Comment by AloisManas** [comment]
Here a screenshot from Youtube, supposedly captured on PS2:
![sh2_youtube_01](https://user-images.githubusercontent.com/53020595/61414863-606e4400-a8ef-11e9-9fa9-ff5d2cc7ec90.png)

https://youtu.be/VUnvAuauSW4?t=11857

**Comment by Polymega** [comment]
Thanks for the screenshot/link! Here's another plot twist: Using the newest version of PCSX2 1.5.0 I can also set "Accurate Date" and get the improved outdoor shadows that you see, but it doesn't work for Maria's ending: 

![image](https://user-images.githubusercontent.com/27395260/61416454-bd530580-a8c1-11e9-9869-e967f0ea0a94.png)

Weird and ultimately not a big deal (this camera cut/shadow is very brief). In any case, we can add this to our feature list as something to hopefully improve in the future. Thanks for providing info about the Accurate Date thing. Something else interesting from this: 

![image](https://user-images.githubusercontent.com/27395260/61416531-f8edcf80-a8c1-11e9-91bc-ae582cd8dd63.png)

The bottom line reads "Full mode is not available on Direct3D." I tested "Fast" mode on PCSX2 via Direct3D 11 (hardware) mode and it still looked fine. (SH2 PC uses DirectX 8/9.)

**Comment by umbrellacorp53** [comment]
I also was curious about the shadow at the beginning as its not a great first impression. Maybe it's possible to tone the shadow down on the observation deck. The problem is identical with the soft shadows test files.

**Comment by Polymega** [comment]
Okay, I think I know why I thought this was normal behavior; because it is... kind of. 

The PC default outdoor shadow behavior also looks like this whenever you're in East and West town. Take a look at the Motel using the latest build of PCSX2 1.5.0:

![image](https://user-images.githubusercontent.com/27395260/61417644-cf36a780-a8c5-11e9-896f-52e7912997e8.png)

I pulled out my PS2 and, yup, sure enough the shadows look like this when out in the town during daytime. So this shadow type only turns "solid" on the Observation Deck and both rooms for the Eddie boss fight. There might be some other rooms, but I haven't come across them yet.

**Comment by AloisManas** [comment]
You're right, the shadows do look different in other areas and are no longer "solid". However, I tried the OpenGL software renderer and suddenly the shadows look much smoother and better. It's not just because of the low resolution, I tried switching between the software and hardware renderers multiple times, both running at native PS2 resolution and with the software renderer it definitely looks much better.

**Comment by AloisManas** [comment]
I took a better screenshot with the OpenGL software renderer that shows better that the shadows are not glitched in this mode:

![pcsx2 2019-07-18 19-55-47-49](https://user-images.githubusercontent.com/53020595/61480517-6c611100-a996-11e9-91c6-3455ac5436dd.png)

**Comment by Polymega** [comment]
Great, thanks for scoping these areas out. We can get FrozenFish and other's thoughts on this in the near future and see if this shadows type can also get prettier along with the other shadow work we'll be getting in to.

**Comment by AloisManas** [comment]
Thank you so much for you hard work.

Just to be clear, this is not the same "solid" shadow type like the one you see on the Observation Deck, although it kind of looks that way in the screenshot. The shadow seems to be getting darker towards James' feet.

**Comment by Polymega** [comment]
> The shadow seems to be getting darker towards James' feet.

Good to know. For ease of future reference I'm going to call this shadow type (whether it's solid or not) "outdoor shadow" simply because that's when we see it (with the exception of the Eddie boss fight rooms). 

So, from what I understand so far, the outdoor shadow is only solid for the observation deck and Eddie boss rooms? 

It does the fake depth shadow effect for the forest trek and East & West outdoor daytime town segments? 

May I ask you to check the boat dock (leaving Eddie boss fight) and outside front entrance of the Hotel?

**Comment by AloisManas** [comment]
I don't know to be honest, I only tested the beginning of the game and it's been years since I last played the game. All I know is that the observation deck and Eddie boss room use the solid type and the whole path between observation deck and the first monster uses the second outdoor non-solid shadow type.

I can continue playing and will keep taking screenshots, or if you want me to check the areas you mentioned immediately, could you please share a memory card file that has saves close to those locations?

**Comment by Polymega** [comment]
Sure, here are PCSX2 memory card save files:

https://github.com/elishacloud/dxwrapper/issues/9#issuecomment-403179621

**Comment by AloisManas** [comment]
Thanks!

**Comment by AloisManas** [comment]
Both areas are using the non-solid outdoor type:
![pcsx2 2019-07-18 21-37-19-58](https://user-images.githubusercontent.com/53020595/61487521-6a9f4980-a9a6-11e9-8ad9-f5f58886ed94.png)
![pcsx2 2019-07-18 21-38-26-76](https://user-images.githubusercontent.com/53020595/61487531-6f63fd80-a9a6-11e9-9683-2fb9031c921f.png)
![pcsx2 2019-07-18 21-46-17-68](https://user-images.githubusercontent.com/53020595/61487626-a6d2aa00-a9a6-11e9-9b3f-56794157374b.png)
![pcsx2 2019-07-18 21-48-42-65](https://user-images.githubusercontent.com/53020595/61487688-d1bcfe00-a9a6-11e9-8ba7-fd96fcf60a30.png)
![pcsx2 2019-07-18 21-50-22-63](https://user-images.githubusercontent.com/53020595/61487740-e8fbeb80-a9a6-11e9-9a41-92367b57547a.png)

**Comment by Polymega** [comment]
Angela's apartment James has a solid "outdoor" type shadow: 

![image](https://user-images.githubusercontent.com/27395260/62754175-c0629100-ba3c-11e9-8f5b-00c84bc9c0d9.png)
